test/e2e/broker_defaults_webhook_test.go

```diff
@@ -0,0 +1,178 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"knative.dev/eventing/pkg/apis/config"
+	eventingduck "knative.dev/eventing/pkg/apis/duck/v1"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/pkg/system"
+)
+
+func TestBrokerNamespaceDefaulting(t *testing.T) {
+	const (
+		brokerClass = "MTChannelBasedBroker"
+	)
+	ctx := context.Background()
+
+	c := testlib.Setup(t, true)
+	defer testlib.TearDown(c)
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		t.Log("Updating defaulting ConfigMap")
+
+		cm, err := c.Kube.CoreV1().ConfigMaps(system.Namespace()).Get(ctx, config.DefaultsConfigName, metav1.GetOptions{})
+		assert.Nil(t, err)
+
+		// Preserve existing namespace defaults.
+		defaults := make(map[string]map[string]interface{})
+		err = yaml.Unmarshal([]byte(cm.Data[config.BrokerDefaultsKey]), defaults)
+		assert.Nil(t, err)
+
+		defaults["namespaceDefaults"][c.Namespace] = map[string]interface{}{
+			"brokerClass": brokerClass,
+			"delivery": map[string]interface{}{
+				"retry":         5,
+				"backoffPolicy": "exponential",
+				"backoffDelay":  "PT0.5S",
+			},
+		}
+
+		b, err := yaml.Marshal(defaults)
+		assert.Nil(t, err)
+
+		cm.Data[config.BrokerDefaultsKey] = string(b)
+
+		cm, err = c.Kube.CoreV1().ConfigMaps(system.Namespace()).Update(ctx, cm, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+
+		b, err = yaml.Marshal(cm.Data[config.BrokerDefaultsKey])
+		if err != nil {
+			t.Log("error", err)
+		} else {
+			t.Log("CM updated - new values:", string(b))
+		}
+
+		return nil
+	})
+	assert.Nil(t, err)
+
+	// Create a Broker and check whether it has the DeliverySpec set as we specified above.
+	// Since the webhook receives the updates at some undetermined time after the update to reduce flakiness retry after
+	// a delay and check whether it has the shape we want it to have.
+
+	namePrefix := "xyz"
+	n := 0
+	lastName := ""
+
+	backoff := wait.Backoff{
+		Duration: time.Second,
+		Factor:   1.0,
+		Jitter:   0.1,
+		Steps:    5,
+	}
+
+	err = retry.OnError(backoff, func(err error) bool { return err != nil }, func() error {
```

Comment:
ditto here.

Reply:
I did the refactor and was able to run the test successfully locally. 
However, noticed the duration was 1s previously and the default duration in the RetryTestErrors's Backoff is 10ms. Not sure if that may have a detrimental impact ?

Reply:
Oh snap, I'm so sorry. I added this, then kept on reading and FAILED to remove this ditto here.